src/uct/api/uct_def.h

```diff
@@ -79,6 +79,19 @@ typedef struct uct_ep_addr       uct_ep_addr_t;
  * @}
  */
 
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Structure for scatter/gather I/O.
+ *
+ */
+typedef struct uct_iov {
+    void     *addr;  /**< Pointer to data. */
+    size_t    len;   /**< Length of data. */
+    uct_mem_h memh;  /**< Local memory key descriptor for the data */
```

Comment:
according to a comment from @shamisp - let's also add the following fields to that structure, which would be unused for now, to support datatypes architecture:
`size_t stride;`
`unsigned count; `